tests/integration/test_usersync.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm usersync integration test."""
+
+import logging
+
+import pytest
+from helpers import (
+    APP_NAME,
+    LDAP_NAME,
+    METADATA,
+    USERSYNC_NAME,
+    get_memberships,
+    get_unit_url,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.usefixtures("deploy")
+class TestUserSync:
+    """Integration test Ranger usersync."""
+
+    async def test_user_sync(self, ops_test: OpsTest):
+        """Validate users and groups have been synchronized from LDAP."""
+        await ops_test.model.deploy(LDAP_NAME, channel="edge")
+
+        ranger_config = {
+            "charm-function": "usersync",
+        }
+
+        charm = await ops_test.build_charm(".")
+        resources = {
+            "ranger-image": METADATA["resources"]["ranger-image"][
+                "upstream-source"
+            ]
+        }
+        action = (
+            await ops_test.model.applications[LDAP_NAME]
+            .units[0]
+            .run_action("load-test-users")
+        )
+        await action.wait()
+
+        await ops_test.model.deploy(
+            charm,
+            resources=resources,
+            application_name=USERSYNC_NAME,
+            num_units=1,
+            config=ranger_config,
+        )
+
+        await ops_test.model.integrate(USERSYNC_NAME, LDAP_NAME)
+        await ops_test.model.set_config({"update-status-hook-interval": "5m"})
```

Comment:
Are you purposefully resetting to `5m` here for a reason? This will make the tests run a bit slower.

Reply:
No, I was just testing something. Removed now.